src/veh_interact.cpp

```diff
@@ -680,67 +680,176 @@ bool veh_interact::do_install( std::string &msg )
+                 part.has_flag( "WATCH" ) ||
+                 part.has_flag( "ALARMCLOCK" ) ||
+                 part.has_flag( VPFLAG_RECHARGE ) ||
+                 part.has_flag( "VISION" ) ||
+                 part.has_flag( "POWER_TRANSFER" ) ||
+                 part.has_flag( "FAUCET" ) ||
+                 part.has_flag( "STEREO" ) ||
+                 part.has_flag( "CHIMES" ) ||
+                 part.has_flag( "MUFFLER" ) ||
+                 part.has_flag( "REMOTE_CONTROLS" ) ||
+                 part.has_flag( "CURTAIN" ) ||
+                 part.has_flag( "SEATBELT" ) ||
+                 part.has_flag( "SECURITY" ) ||
+                 part.has_flag( "SEAT" ) ||
+                 part.has_flag( "BED" ) ||
+                 part.has_flag( "DOOR_MOTOR" ) ||
+                 part.has_flag( "WATER_PURIFIER" );
+      }
+
+
+
+
+  };
+    // Hull
+    tab_filters[4] = [&]( const vpart_info *p ) {    auto &part = *p;
+        if( !veh->is_furniture ) {
+            return( ( part.has_flag( VPFLAG_OBSTACLE ) ||
+                      part.has_flag( "ROOF" ) ||
+                      part.has_flag( VPFLAG_ARMOR ) ) &&
+                      !part.has_flag( "WHEEL" ) &&
+                      !tab_filters[3]( p ) ) && !part.has_flag( VPFLAG_FURNITURE );
+        } else {
+            return( part.has_flag( VPFLAG_OBSTACLE ) ||
+                part.has_flag( "ROOF" ) ||
+                part.has_flag( VPFLAG_ARMOR ) ) &&
+                !part.has_flag( "WHEEL" ) &&
+                !tab_filters[3]( p );
+        }    
+    };
+    
+    // Internals
     tab_filters[5] = [&](const vpart_info *p) { auto &part = *p;
-                                                   return part.has_flag(VPFLAG_ENGINE) || // Internals
-                                                   part.has_flag(VPFLAG_ALTERNATOR) ||
-                                                   part.has_flag(VPFLAG_CONTROLS) ||
-                                                   part.location == "fuel_source" ||
-                                                   part.location == "on_battery_mount" ||
-                                                   (part.location.empty() && part.has_flag("FUEL_TANK")); };
+        if( !veh->is_furniture  ) {
+            return ( part.has_flag( VPFLAG_ENGINE ) ||
+              part.has_flag( VPFLAG_ALTERNATOR ) ||
+              part.has_flag( VPFLAG_CONTROLS ) ||
+              part.location == "fuel_source" ||
+              part.location == "on_battery_mount" ||
+              ( part.location.empty() &&part.has_flag( "FUEL_TANK" ) ) ) && !part.has_flag( VPFLAG_FURNITURE );
+        } else {
+            return part.has_flag( VPFLAG_ENGINE ) ||
+            part.has_flag( VPFLAG_ALTERNATOR ) ||
+            part.has_flag( VPFLAG_CONTROLS ) ||
+            part.location == "fuel_source" ||
+            part.location == "on_battery_mount" ||
+            ( part.location.empty() && part.has_flag( "FUEL_TANK" ) );
+        }
```

Comment:
This will need a massive cleanup.

Reply:
Instead of doing this:
```
if( veh_furniture ) {
    if( x && y && z && w ) {
        do stuff;
    }
} else {
    if( x && y && z && w && part_not_furniture ) {
        do stuff;
    }
}
```
You have to do it like this:
```
if( x && y && z && w && ( !veh_furniture || !part_not_furniture ) ) {
    do stuff;
}
```
If you are only adding conditions in an `if`, do not create nested `if`s, but include the conditions in the first `if`.

Reply:
Even better in this case, capture the intermediate value with a boolean since the logic statement is so long already:
```
bool is_internal = part.has_flag( VPFLAG_ENGINE ) || part.has_flag( VPFLAG_ALTERNATOR ) ||
    part.has_flag( VPFLAG_CONTROLS ) || part.location == "fuel_source" ||
    part.location == "on_battery_mount" || ( part.location.empty() && part.has_flag( "FUEL_TANK" ) );
return is_internal && !veh->is_furniture && !part.has_flag( VPFLAG_FURNITURE ) );
```